src/workerd/api/node/node-compat-v2-test.js

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+//
+
+// Imports of Node.js built-ins should work both with and without
+// the 'node:' prefix.
+import { default as assert } from 'node:assert';
+import { default as assert2 } from 'assert';
+const assert3 = (await import('node:assert')).default;
+const assert4 = (await import('assert')).default;
+
+assert.strictEqual(assert, assert2);
+assert.strictEqual(assert, assert3);
+assert.strictEqual(assert, assert4);
+
+export const nodeJsExpectedGlobals = {
+  async test() {
+    // Expected Node.js globals Buffer, process, and global should be present.
+    const { Buffer } = await import('node:buffer');
+    assert.strictEqual(Buffer, globalThis.Buffer);
+
+    const { default: process } = await import('node:process');
+    assert.strictEqual(process, globalThis.process);
+
+    assert.strictEqual(global, globalThis);
+  }
+};
+
+export const nodeJsGetBuiltins = {
+  async test() {
+    // node:* modules in the worker bundle should override the built-in modules...
```

Comment:
Does the same apply to non-`node:` prefixed modules? Do we want tests for it too?

Reply:
It does but tests for that should be added as a separate PR.